Common/include/grid_movement/CRadialBasisFunctionInterpolation.hpp

```diff
@@ -0,0 +1,143 @@
+   * \param[in] UpdateGeo - Update geometry.
+   * \param[in] Derivative - Compute the derivative (disabled by default). Does not actually deform the grid if enabled.
+   */
+  void SetVolume_Deformation(CGeometry* geometry, CConfig* config, bool UpdateGeo, bool Derivative,
+                                                bool ForwardProjectionDerivative);
+  
+  /*! 
+  * \brief Obtaining the interpolation coefficients based on displacement of the control nodes.
+  * \param[in] geometry - Geometrical definition of the problem.
+  * \param[in] config - Definition of the particular problem.
+  * \param[in] iNonlinear_Iter - Surface deformation step.
+  */
+  void GetInterpolationCoefficients(CGeometry* geometry, CConfig* config, unsigned long iNonlinear_Iter);
+
+  /*!
+  * \brief Compute the interpolation matrix with Radial Basis Function evaluations.
+  * \param[in] geometry - Geometrical definition of the problem.
+  * \param[in] config - Definition of the particular problem.
+  */
+  void SetInterpolationMatrix(CGeometry* geometry, CConfig* config);
+
+  /*!
+  * \brief Build the deformation vector with surface displacements of the control nodes.
+  * \param[in] geometry - Geometrical definition of the problem.
+  * \param[in] config - Definition of the particular problem.
+  */
+  void SetDeformationVector(CGeometry* geometry, CConfig* config);
+
+  /*!
+  * \brief Selecting unique set of control nodes based on marker information.
+  * \param[in] geometry - Geometrical definition of the problem.
+  * \param[in] config - Definition of the particular problem.
+  */
+  void SetControlNodes(CGeometry* geometry, CConfig* config);
+
+  /*!
+  * \brief Selecting internal nodes for the volumetric deformation.
+  * \param[in] geometry - Geometrical definition of the problem.
+  */
+  void SetInternalNodes(CGeometry* geometry);
+
+  /*!
+  * \brief Solving of the Radial Basis Function interpolation system, yielding the interpolation coefficients
+  */
+  void SolveRBF_System(void);
+
+  
+  /*!
+  * \brief Updating the grid coordinates.
+  * \param[in] geometry - Geometrical definition of the problem.
+  * \param[in] config - Definition of the particular problem.
+  */
+  void UpdateGridCoord(CGeometry* geometry, CConfig* config);
+
+  /*!
+  * \brief Custom comparison function, for sorting the CRadialBasisFunctionNode objects based on their index.
+  * \param[in] a - First considered Radial Basis Function Node.
+  * \param[in] b - second considered Radial Basis Function Node.
+  */
+  inline static bool Compare(CRadialBasisFunctionNode* a, CRadialBasisFunctionNode* b){
+    return a->GetIndex() < b->GetIndex();
+  }
+
+  /*!
+  * \brief Custom equality function, for obtaining a unique set of CRadialBasisFunctionNode objects.
+  * \param[in] a - First considered Radial Basis Function Node.
+  * \param[in] b - second considered Radial Basis Function Node.
+  */
+  inline static bool Equal(CRadialBasisFunctionNode* a, CRadialBasisFunctionNode* b){
```

Comment:
Same above with consistent naming convention of overall code. Maybe @pcarruscag have a better suggestion

Reply:
Also here I have changed it to `HasEqualIndex`.